src/aspire/estimation/covar2d.py

```diff
@@ -9,25 +9,107 @@
 
 from aspire.utils.matrix import shrink_covar
 from aspire.utils.optimize import conj_grad
-from aspire.denoise.covar2d import RotCov2D
-from aspire.utils import ensure
 
+from aspire.utils import ensure
 
 logger = logging.getLogger(__name__)
 
 
-class Cov2DCTF(RotCov2D):
+class RotCov2D:
     """
-    Define a derived class for denoising 2D images using CTF information and the Covariance Wiener Filtering (CWF)
-    Cov2D method described in
+    Define a class for denoising 2D images using the Cov2D method with CTF information described in
 
     T. Bhamre, T. Zhang, and A. Singer, "Denoising and covariance estimation of single particle cryo-EM images",
     J. Struct. Biol. 195, 27-81 (2016). DOI: 10.1016/j.jsb.2016.04.013
     """
 
+    def __init__(self, basis, as_type='single'):
+        """
+        constructor of an object for 2D covariance analysis
+        """
+        self.basis = basis
+        ensure(basis.d == 2, 'Only two-dimensional basis functions are needed.')
+        self.as_type = as_type
```

Comment:
Why does this have to be specified here? Shouldn't the type be inferred from the input (`coeffs`)? Also, it seems like it's only used in `get_mean`.

Reply:
This initializes a default data type for calculating mean and covariance,  and also sets the data type for block diagonal matrix. Add the statement to get the inferred value from `coeffs`.

Reply:
I still don't understand why this needs to be stored as an attribute. Couldn't `get_mean` and `get_covar` just use the data type of `coeffs`?